acceptance/tests/language/invoke_pp_function_from_multiple_environments.rb

```diff
@@ -0,0 +1,61 @@
+test_name 'Puppet-language functions can be called from any environment' do
+
+  mini_func = 'function jenny::mini($a, $b) {if $a <= $b {$a} else {$b}}'
+  maxi_func = 'function jenny::nested::maxi($a, $b) {if $a >= $b {$a} else {$b}}'
+  step 'Create PP functions on master' do
+    manifest = <<-EOM.gsub(/^ {6}/,'')
+      File {
+        owner => 'puppet',
+        group => 'puppet',
+        ensure => present,
+        mode => '0644',
+      }
+      file {['/etc/puppetlabs',
+          '/etc/puppetlabs/code',
+          '/etc/puppetlabs/code/modules',
+          '/etc/puppetlabs/code/modules/jenny',
+          '/etc/puppetlabs/code/modules/jenny/functions',
+          '/etc/puppetlabs/code/modules/jenny/functions/nested',
+          '/etc/puppetlabs/code/environments',
+          '/etc/puppetlabs/code/environments/frick',
+          '/etc/puppetlabs/code/environments/frack',]:
+        ensure => directory,
+        mode => '0755',
+      }
+      file {'/etc/puppetlabs/code/modules/jenny/functions/mini.pp':
+        content => '#{mini_func}',
+      }
+      file {'/etc/puppetlabs/code/modules/jenny/functions/nested/maxi.pp':
+        content => '#{maxi_func}',
+      }
+    EOM
+    apply_manifest_on(master, manifest)
+  end
+
+  applyme = <<-EOM
+    class f {
+      notify { 'mini': message => jenny::mini(1,2), }
+      notify { 'maxi': message => jenny::nested::maxi(1,2), }
+    }
+    include 'f'
+  EOM
+
+  step 'Call a PP function from two environments' do
+    env_1 = on master, puppet("apply -e \"#{applyme}\" --environment frick")
```

Comment:
we should probably eliminate as many puppet apply calls as possible (see runtime of this on windows).  applyme manifest could be merged with the setup manifest above it.  if you don't want to repeat yourself with the two environments you could abstract it into a function to replace the environment name in the two environment sections.
thoughts?